### PR TITLE
Support Testnet4 Network

### PR DESCRIPTION
**Pull request by will-bitlight**
This PR integrates the latest Network changes to support `Testnet4`, ensuring compatibility with the `Resolver` in `bp-wallet` and adapting the `RgbResolver` within RGB. 

According to our past PR experience, it has been modified and optimized

If you have any suggestions, please leave your valuable comments, I will actively cooperate with the revision, thank you very much

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -29,14 +29,14 @@ commit_verify = "0.11.0-beta.6"
 bp-core = "0.11.0-beta.6"
 bp-seals = "0.11.0-beta.6"
 bp-std = "0.11.0-beta.6"
-bp-electrum = "0.11.0-beta.6"
-bp-esplora = { version = "0.11.0-beta.6", default-features = false, features = ["blocking"] }
```
Comment (dr-orlovsky) [inline Cargo.toml:33]: Why do we re-order crates here?

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]
LGTM except some dependency specs (pls see comments)

File: Cargo.toml (hunk 1 above)
Comment (will-bitlight) [reply Cargo.toml:33]: Thank you for the reminder. I will restore the order of the crates shortly.

During development, I centralized the crates that needed to be referenced via git, which led to changes in the order.

File: Cargo.toml [hunk 2]
```diff
@@ -29,14 +29,14 @@ commit_verify = "0.11.0-beta.6"
...
 descriptors = "0.11.0-beta.6"
 psbt = { version = "0.11.0-beta.6", features = ["client-side-validation"] }
-bp-wallet = { version = "0.11.0-beta.6.1" }
+bp-wallet = { version = "0.11.0-beta.6.1", features = ["cli"] }
```
[the remark this answers is not in the document]
Comment (will-bitlight) [reply Cargo.toml:34]: Yes, this feature should be introduced in the minimal way. I will make the adjustments shortly.

**Comment by will-bitlight** [comment]
Thank you very much, I think the code changes are much clearer now.

**Review by dr-orlovsky — APPROVED** [review]

**Pull request merged**